### ISSUE TITLE
Test failures in TestPerm16 and TestPermAll

### ISSUE THREAD

**Issue by thierry-FreeBSD** [hivert/HPCombi#21]
On FreeBSD, with Clang 16.0.6 and Boost 1.83.0, 2 errors occur during the tests of 0.0.7:
```
Test project /usr/ports/math/hpcombi/work/.build
    Start 2: TestPerm16
1/2 Test #2: TestPerm16 .......................***Failed    2.82 sec
Running 47 test cases...
unknown location(0): fatal error: in "PTransf16_test/PTransf16_rank_ref": signal: SIGABRT (application abort requested)
/usr/ports/math/hpcombi/work/HPCombi-0.0.7/tests/test_perm16.cpp(203): last checkpoint

*** 1 failure is detected in the test module "Perm16Tests"


    Start 3: TestPermAll
2/2 Test #3: TestPermAll ......................***Failed    0.38 sec
Running 133 test cases...
unknown location(0): fatal error: in "VectType_test/operator_bracket_const_test<Fixture<HPCombi__PermGeneric<12_ unsigned char>>>": signal: SIGABRT (application abort requested)
/usr/ports/math/hpcombi/work/HPCombi-0.0.7/tests/test_perm_all.cpp(105): last checkpoint
unknown location(0): fatal error: in "VectType_test/operator_bracket_test<Fixture<HPCombi__PermGeneric<12_ unsigned char>>>": signal: SIGABRT (application abort requested)
/usr/ports/math/hpcombi/work/HPCombi-0.0.7/tests/test_perm_all.cpp(114): last checkpoint

*** 2 failures are detected in the test module "perm_generic"



0% tests passed, 2 tests failed out of 2

Total Test time (real) =   3.21 sec

The following tests FAILED:
          2 - TestPerm16 (Failed)
          3 - TestPermAll (Failed)
Errors while running CTest

```

**Comment by james-d-mitchell** [comment]
Thanks for the report @thierry-FreeBSD, will try to take a look tomorrow

**Comment by hivert** [comment]
Thanks for the report. I should have caught those with some sanitize tools (thanks @james-d-mitchell for helping me setup those).

Should be fixed now in master!

**Comment by hivert** [comment]
Fixed in V0.0.9 release ! Thanks again.
